MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -461,214 +500,251 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputCloseSurfaceModel( vtkMRMLMarkups
+void vtkSlicerMarkupsToModelLogic::ComputeTransformedExtentRanges( vtkPoints* points, vtkMatrix4x4* transformMatrix, double outputExtentRanges[ 3 ] )
 {
-  double axisLength = vtkMath::Norm( axis );
-  bool axisLengthEqualToOne = abs( axisLength - 1.0 ) < COMPARE_TO_ZERO_TOLERANCE;
-  if ( !axisLengthEqualToOne )
+  if ( points == NULL )
   {
-    vtkWarningMacro( "Axis " << axis[ 0 ] << ", " << axis[ 1 ] << ", " << axis[ 2 ] << " "
-                  << "provided to ComputePointRangeAlongAxis() does not have unit length. "
-                  << "Unable to determine whether points are coplanar." );
-    return false;
+    vtkErrorMacro( "points is null. Aborting output extent computation." );
+    return;
   }
   
-  // determine the range of point distances along this axis
-  double pointRangeMin = 0.0; // temporary value
-  double pointRangeMax = 0.0; // temporary value
-  int numberOfPoints = points->GetNumberOfPoints();
-  for ( int i = 0; i < numberOfPoints; i++ )
-  {
-    double currentPoint[ 3 ] = { 0.0, 0.0, 0.0 };
-    points->GetPoint( i, currentPoint );
-    double pointDistanceAlongAxis = vtkMath::Dot( currentPoint, axis );
-    if ( i == 0 )
-    {
-      // the first point will establish the initial maximum *and* minimum distance along the axis
-      pointRangeMin = pointDistanceAlongAxis;
-      pointRangeMax = pointDistanceAlongAxis;
-    }
-    else
-    {
-      // determine whether this point extends the range along the axis
-      if ( pointDistanceAlongAxis < pointRangeMin )
-      {
-        pointRangeMin = pointDistanceAlongAxis;
-      }
-      if ( pointDistanceAlongAxis > pointRangeMax )
-      {
-        pointRangeMax = pointDistanceAlongAxis;
-      }
-    }
+  if ( transformMatrix == NULL )
+  {
+    vtkErrorMacro( "transformMatrix is null. Aborting output extent computation." );
+    return;
+  }
+  
+  if ( outputExtentRanges == NULL )
+  {
+    vtkErrorMacro( "outputExtentRanges is null. Aborting output extent computation." );
+    return;
+  }
+
+  vtkSmartPointer< vtkTransform > transform = vtkSmartPointer< vtkTransform >::New();
+  transform->SetMatrix( transformMatrix );
+  transform->Update();
+
+  // can't transform points directly, so need to store in a container
+  vtkSmartPointer< vtkPolyData > polyDataWithPoints = vtkSmartPointer< vtkPolyData >::New();
+  polyDataWithPoints->SetPoints( points );
+
+  vtkSmartPointer< vtkTransformFilter > transformFilter = vtkSmartPointer< vtkTransformFilter >:: New();
+  transformFilter->SetTransform( transform );
+  transformFilter->SetInputData( polyDataWithPoints );
+  transformFilter->Update();
+
+  // the extent can be extracted from the output points object (poly data bounds does not work)
+  vtkPoints* transformedPoints = transformFilter->GetPolyDataOutput()->GetPoints();
+  transformedPoints->ComputeBounds();
+  double* extents = transformedPoints->GetBounds(); // { xmin, xmax, ymin, ymax, zmin, zmax }
+
+  double xmin = extents[ 0 ];
```

Comment:
Too verbose. Replace the 4 lines copy-pasted 3x by these 2 lines

for (int axisIndex=0; axisIndex<3; ++axisIndex)
{
  extents[axisIndex]=extents[axisIndex*2+1]-extents[axisIndex*2];
}